multidim_threshold/projection.py

```diff
@@ -63,6 +75,12 @@ def midpoint(lo, hi, proj_vec):
     hi = clip_rec(proj_vec, hi)
     return (np.array(lo) + np.array(hi)) / 2
 
+def random_root(lo, hi):
+    dim = len(lo)
+    t_proj = lambda lo, hi, i: lo[i] + (hi[i] - lo[i]) * random.uniform(0, 1)
```

Comment:
Can now drop the `lo, hi` arguments since they'll be closed over:

```python
 t_proj = lambda i: lo[i] + (hi[i] - lo[i]) * random.uniform(0, 1)
```